### PR TITLE
Heroku auto deployment

### LINKED ISSUE #53: One click Heroku app deployment button
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/meilix

**Expected Behaviour**
A button for one click Heroku app deployment

### LINKED ISSUE #57: Python Package Version Pinning
- [X] bug report
- [X] feature request

If we want to install the program in one year in the uture, the required packages should be pinned to assure that the API changes in the installed packages do not break the program.
Further Reading: https://github.com/AllYarnsAreBeautiful/ayab-desktop/issues/56

Suggestion:
- You can use the package pip-tools to pin the versions of all packages and those packages required by them.

### PR DESCRIPTION
**Pull request by meets2tarun**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [ ] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes (provide a screenshot or link for test) 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
One can use the button to deploy the app on heroku with just one click

#### Changes proposed in this pull request:

- Button for one-click deployment (edited /doc/installation/heroku.md)
- app.json to execute that

Fixes #53 
Fixes #57

### CODE REVIEW

File: docs/installation/my_token.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ It contain 3 words as :
 This is the user whose branch of that repository is going to be trigger.
 We are using [fossasia/meilx](https://github.com/fossasia/meilix) repository to trigger its build.
 
-**Before undergoing any development process you must fork [fossasia/meilx](https://github.com/fossasia/meilix) repository and change the username so that the release will be done in your repository**
+**Before undergoing any development process you must fork [fossasia/meilx](https://github.com/fossasia/meilix) repository and change the username so that the release will be done in your repository**.
 You can fork [this](https://github.com/fossasia/meilix) and start to put **your username** so that the fossasia/meilix repo will not get flooded with unnecessary builds. And your can easily play with configuration without disturbing the original repo.
```
Comment (yashLadha) [inline docs/installation/my_token.md]: It would be great if we specify the file in which we have to change it. So that there is no confusion.

Comment (meets2tarun) [inline docs/installation/my_token.md]: It is very well mentioned on the 5th line.

Comment (yashLadha) [inline docs/installation/my_token.md]: Yups. :+1:

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,89 @@
-Flask
-gunicorn
-werkzeug
+attrs==15.2.0
```
Comment (meets2tarun) [inline requirements.txt]: @niccokunzmann Your views are python package version pinning?
https://github.com/fossasia/meilix-generator/issues/57

File: requirements.txt [hunk 3]
```diff
@@ -1,3 +1,89 @@
...
+pycrypto==2.6.1
+pycups==1.9.73
+pycurl==7.43.0
+Pygments==2.1
+pygobject==3.20.0
+pyinotify==0.9.6
+PyJWT==1.3.0
+pyOpenSSL==0.15.1
+pyparted==3.10.7
+pyserial==3.0.1
```
Comment (niccokunzmann) [inline requirements.txt]: I am not sure you need this package.
These are a lot of packages, potentially all that are installed on your system. I conclude, you use Ubuntu as your operating system.

The real list should be much smaller.
Example repository:
- https://github.com/schul-cloud/schul_cloud_search_tests
The idea is for you as a developer, you can do the following:
1. `pip install --upgrade -r requirements.in` installs latest packages
2. run tests
3. `pip-compile --output-file requirements.txt requirements.in` fix the versions that definitely work for an eternity.

Comment (niccokunzmann) [inline requirements.txt]: The list should look like this with the == signs but it must be smaller. I guess, you used pip freeze to geenrate the list.

**Comment by abishekvashok** [comment]
Check Travis

File: .travis.yml [hunk 4]
```diff
@@ -10,4 +10,4 @@ deploy:
   on:
     repo: fossasia/meilix-generator
     branch:
-    - master
+    - master
```
Comment (abishekvashok) [inline .travis.yml]: Always add a newline at EOF

**Comment by xeon-zolt** [comment]
#66

File: app.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        "FORCE_SSL": "true",
+        "INVITATION_CODE": {
+            "generator": "secret"
+        }
+    },
+    "buildpacks": [
+        {
+            "url": "heroku/python"
+        },
+    ],
```
Comment (xeon-zolt) [inline app.json]: remove `,`

**Comment by niccokunzmann** [comment]
@meets2tarun @xeon-zolt Can I help you merge the pull-requests in any way? Huge open pull-requests can slow down the development process. This is why PRs should be small and addressing one issue/problem only. Then, they are independent and do not slow down each other/other contributions through conflicts. What do you think can be done?

**Comment by meets2tarun** [comment]
@niccokunzmann Please approve the PR if you found that there is everything OK.
Then we'll merge it. 
Thanks :)

**Comment by abishekvashok** [comment]
@meets2tarun pls squash the commits with some nice commit messages. See the Linux repo on github as a reference :)

File: app.json [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        },
+        "ON_HEROKU": "true",
+        "FORCE_SSL": "true",
+        "INVITATION_CODE": {
+            "generator": "secret"
+        }
+    },
+    "buildpacks": {
+            "url": "heroku/python"
+        }
```
Comment (abishekvashok) [inline app.json]: Properly intend it pls.

**Comment by meets2tarun** [comment]
My working directory is not in sync with the local directory. It may create lots of conflicts while doing that.
Don't worry about the squashing, it can also be done while merging with a good commit message.
Please review and approve the code. @abishekvashok @xeon-zolt

**Pull request merged**